### DISCUSSION TITLE
How to Restart strongSwan on AUTH_FAILED Received from Remote Server?

### DISCUSSION THREAD

**Discussion by exoticprgmr** [strongswan/strongswan#2579] [General]
Hello everyone,

I am trying to configure strongSwan to automatically restart when it receives an AUTH_FAILED response from the remote server. So far, I haven't been successful despite trying various configuration options.

Here’s the issue I am facing:

I use strongSwan to connect to Proton VPN. If my gateway restarts without properly disconnecting from the Proton VPN server, the server throws an AUTH_FAILED error until the connection is cleared on Proton side. In these cases, I need to restart the strongSwan service manually to resolve the issue.

Is there a way to configure strongSwan to handle this situation automatically? Ideally, I’d like it to restart or reinitialize the connection whenever it encounters AUTH_FAILED.

Any suggestions or pointers would be greatly appreciated! I use debian bookworm with strongswan 5.9.8 

Thanks in advance for your help!

Logs:
Dec 19 22:20:46 charon-systemd[2363]: authentication of 'xx.xx.xx.xx' with RSA_EMSA_PKCS1_SHA2_384 successful
Dec 19 22:20:46 charon-systemd[2363]: server requested EAP_IDENTITY (id 0x00), sending 'username'
Dec 19 22:20:46 charon-systemd[2363]: generating IKE_AUTH request 2 [ EAP/RES/ID ]
Dec 19 22:20:46 charon-systemd[2363]: sending packet: from MYIP[4500] to PROTONIP[4500] (82 bytes)
Dec 19 22:20:50 charon-systemd[2363]: retransmit 1 of request with message ID 2
Dec 19 22:20:50 charon-systemd[2363]: sending packet: from MYIP[4500] to PROTONIP[4500] (82 bytes)
Dec 19 22:20:52 charon-systemd[2363]: received packet: from PROTONIP[4500] to MYIP[4500] (65 bytes)
Dec 19 22:20:52 charon-systemd[2363]: parsed IKE_AUTH response 2 [ EAP/FAIL ]
Dec 19 22:20:52 charon-systemd[2363]: received EAP_FAILURE, EAP authentication failed
Dec 19 22:20:52 charon-systemd[2363]: generating INFORMATIONAL request 3 [ N(AUTH_FAILED) ]
Dec 19 22:20:52 charon-systemd[2363]: sending packet: from MYIP[4500] to PROTONIP[4500] (65 bytes)

**Comment by tobiasbrunner** [comment]
> Is there a way to configure strongSwan to handle this situation automatically?

Not directly. One possibility is via trap policies to retrigger an acquire on matching outbound traffic (note the [`charon.plugins.kernel-netlink.xfrm_acq_expires`](https://docs.strongswan.org/docs/latest/config/strongswanConf.html#_charon_plugins_kernel_netlink) option, though). Otherwise, you have to script it via [vici](https://docs.strongswan.org/docs/latest/plugins/vici.html) (and possibly the [error-notify](https://docs.strongswan.org/docs/latest/plugins/error-notify.html) plugin).
